src/pip/_internal/utils/misc.py

```diff
@@ -132,14 +132,58 @@ def get_prog():
     return 'pip'
 
 
-# Retry every half second for up to 3 seconds
-@retry(stop_max_delay=3000, wait_fixed=500)
 def rmtree(dir, ignore_errors=False):
     # type: (str, bool) -> None
+    """remove directory tree, with appropriate retry logic per OS."""
+    rmtree_impl(dir, ignore_errors)
+
+
+@retry(stop_max_delay=3000, wait_fixed=500)
+def rmtree_minimal_retry(dir, ignore_errors=False):
+    # type: (str, bool) -> None
+    """remove directory tree, retry up to 3 seconds."""
     shutil.rmtree(dir, ignore_errors=ignore_errors,
                   onerror=rmtree_errorhandler)
 
 
+# default rmtree implementation
+rmtree_impl = rmtree_minimal_retry
+# on win32, wrap rmtree with more retries.
+# this helps remediate the case when another when a virus scanner (or any
+# other process) is still holding a handle to the file being deleted.
+if sys.platform == "win32":
+    if sys.version_info >= (3,):
+        def is_locked(e):
+            """on windows python 3, returns True if the OSError indicates
+            that another process has a handle on the file."""
+            return e.errno == errno.EACCES and e.winerror == 5
+    else:
+        def is_locked(e):
+            """on windows python 2.7, returns True if the OSError indicates
+            that another process has a handle on the file."""
+            return e.errno == 41 and e.winerror == 145
```

Comment:
41 is defined in the C standard, and can be accessed with `errno.EPROTOTYPE`. 145 is a Windows error defined in `WinError.h` as `ERROR_DIR_NOT_EMPTY`, but Python’s stdlib does not expose this value AFAIK.

Reply:
I've used EACCES instead of 13. but errno=41 does not have a constant associated with it. for the windows constants i provided a URL reference since the constants aren't exposed anywhere I could find.